From 2e6f25d43dcbdaa56f64f8fc657bd81eddd28ac7 Mon Sep 17 00:00:00 2001
From: Eric Sandeen <sandeen@redhat.com>
Date: Mon, 4 Aug 2014 18:58:39 -0400
Subject: [fs] btrfs: make btrfs_debug match pr_debug handling related to DEBUG

Message-id: <1407179109-13714-33-git-send-email-sandeen@redhat.com>
Patchwork-id: 87346
O-Subject: [RHEL7.1 PATCH 032/422] Btrfs: make btrfs_debug match pr_debug handling related to DEBUG
Bugzilla: 1071179
RH-Acked-by: Zach Brown <zab@redhat.com>

From: Frank Holton <fholton@gmail.com>

commit 27a0dd61a5cf742ebcd7a82c47be2502b1113eff
Author: Frank Holton <fholton@gmail.com>
Date:   Tue Nov 12 19:22:53 2013 -0500

    Btrfs: make btrfs_debug match pr_debug handling related to DEBUG

    The kernel macro pr_debug is defined as a empty statement when DEBUG is
    not defined. Make btrfs_debug match pr_debug to avoid spamming
    the kernel log with debug messages

    Signed-off-by: Frank Holton <fholton@gmail.com>
    Signed-off-by: Josef Bacik <jbacik@fusionio.com>
    Signed-off-by: Chris Mason <clm@fb.com>
Signed-off-by: Eric Sandeen <sandeen@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/fs/btrfs/ctree.h b/fs/btrfs/ctree.h
index 47835f5..7158c97 100644
--- a/fs/btrfs/ctree.h
+++ b/fs/btrfs/ctree.h
@@ -3816,8 +3816,14 @@ void btrfs_printk(const struct btrfs_fs_info *fs_info, const char *fmt, ...)
  btrfs_printk(fs_info, KERN_NOTICE fmt, ##args)
 #define btrfs_info(fs_info, fmt, args...) \
  btrfs_printk(fs_info, KERN_INFO fmt, ##args)
+
+#ifdef DEBUG
 #define btrfs_debug(fs_info, fmt, args...) \
  btrfs_printk(fs_info, KERN_DEBUG fmt, ##args)
+#else
+#define btrfs_debug(fs_info, fmt, args...) \
+    no_printk(KERN_DEBUG fmt, ##args)
+#endif
 
 #ifdef CONFIG_BTRFS_ASSERT
 
-- 
1.7.1